### PR TITLE
move serde jackson to testImplementation

### LINKED ISSUE #1393: Adding dependency on io.micronaut.security:micronaut-security-oauth2 overrides io.micronaut:micronaut-jackson-databind with Micronaut Serialization
### Expected Behavior

Adding a dependency on `io.micronaut.security:micronaut-security-oauth2` should not enforce me to use Micronaut Serialization. 

We rely on the implicit support for json serialization that Jackon Databind brings in our application. When upgrading to Micronaut 4.0.1 our tests started to fail with error messages similar to this
```
Error encoding object [CustomResponse(aProperty=Hello World)] to JSON: No serializable introspection present for type CustomResponse B. Consider adding Serdeable. Serializable annotate to type CustomResponse B. Alternatively if you are not in control of the project's source code, you can use @SerdeImport(CustomResponse.class) to enable serialization of this type.
```

After some troubleshooting we identified the cause to be that `io.micronaut.serde.jackson.JacksonJsonMapper` is annotated with `@Primary` and that it is brought in as a transitive dependency from `io.micronaut.security:micronaut-security-oauth2`

### Actual Behaviour

Adding the dependency `io.micronaut.security:micronaut-security-oauth2` will bring in the dependency `io.micronaut.serde:micronaut-serde-jackson` effectively overriding `io.micronaut:micronaut-jackson-databind`

### Steps To Reproduce

1. Create a Micronaut project where you want to use io.micronaut:micronaut-jackson-databind
2. Configure a simple data class that is used as a Http response
3. Add a dependency on io.micronaut.security:micronaut-security-oauth2

See the example application linked below

### Environment Information

_No response_

### Example Application

https://github.com/olamelin/micronaut-security-serde/tree/main

### Version

4.0.1

### PR DESCRIPTION
**Pull request by sdelamo**
depend only on jackson annotations

Close #1393

### CODE REVIEW

File: test-suite-http/build.gradle [hunk 1]
```diff
@@ -14,7 +14,7 @@ dependencies {
 
     testImplementation(projects.micronautSecurity)
     testImplementation(projects.micronautSecurityJwt)
-    testImplementation(projects.micronautSecurityOauth2)
+    //testImplementation(projects.micronautSecurityOauth2)
```
Comment (sdelamo) [inline test-suite-http/build.gradle:17]: I will investigate this in a separate PR. It seems Security OAuth is coupled to JsonMapper.

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java [hunk 2]
```diff
@@ -34,48 +33,125 @@
  */
 @ReflectiveAccess
 @Serdeable
-@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Comment (yawkat) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java:37]: serde supports this, this change shouldn't be necessary

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/OpenIdTokenResponse.java [hunk 3]
```diff
@@ -32,9 +31,9 @@
  */
 @ReflectiveAccess
 @Introspected
-@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Comment (yawkat) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/OpenIdTokenResponse.java:35]: same here

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/TokenErrorResponse.java [hunk 4]
```diff
@@ -30,15 +29,16 @@
  * @since 1.2.0
  */
 @Introspected
-@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Comment (yawkat) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/TokenErrorResponse.java:33]: same here

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/TokenResponse.java [hunk 5]
```diff
@@ -35,23 +34,27 @@
  */
 @ReflectiveAccess
 @Introspected
-@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Comment (yawkat) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/TokenResponse.java:38]: same here

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java (hunk 2 above)
Comment (sdelamo) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java:37]: I just wanted to remove even the compile time dependency to mn.jackson.databind

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/AuthorizationCodeGrant.java [hunk 6]
```diff
@@ -31,17 +30,20 @@
  * @since 1.2.0
  */
 @Introspected
-@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Comment (yawkat) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/AuthorizationCodeGrant.java:34]: same here

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/ClientCredentialsGrant.java [hunk 7]
```diff
@@ -33,11 +32,11 @@
  * @since 1.2.0
  */
 @Introspected
-@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Comment (yawkat) [inline security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/ClientCredentialsGrant.java:36]: same here, i'm gonna stop now

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java (hunk 2 above)
Comment (yawkat) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java:37]: why? it should be fine as a compile-time dep, no?

**Review by yawkat — APPROVED** [review]

**Review by yawkat — APPROVED** [review]

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java (hunk 2 above)
Comment (sdelamo) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java]: I don't see much benefit on `JsonNaming` + the extra `compileOnly` and prefer the the explicit `@JsonProperty`. The only class which really benefits from `JsonNaming` is `DefaultOpenIdProviderMetadata`.

Nonetheless, I have reverted that change here: https://github.com/micronaut-projects/micronaut-security/pull/1395/commits/c5637c6e9e7222e3edfa1dafcf1b5bd3c1ec0b65

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/TokenErrorResponse.java (hunk 4 above)
Comment (sdelamo) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/token/response/TokenErrorResponse.java]: I have reverted that change here: https://github.com/micronaut-projects/micronaut-security/pull/1395/commits/c5637c6e9e7222e3edfa1dafcf1b5bd3c1ec0b65

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/AuthorizationCodeGrant.java (hunk 6 above)
Comment (sdelamo) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/AuthorizationCodeGrant.java]: I have reverted that change here: https://github.com/micronaut-projects/micronaut-security/pull/1395/commits/c5637c6e9e7222e3edfa1dafcf1b5bd3c1ec0b65

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/ClientCredentialsGrant.java (hunk 7 above)
Comment (sdelamo) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/grants/ClientCredentialsGrant.java]: I have reverted that change here: https://github.com/micronaut-projects/micronaut-security/pull/1395/commits/c5637c6e9e7222e3edfa1dafcf1b5bd3c1ec0b65

File: security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java (hunk 2 above)
Comment (yawkat) [reply security-oauth2/src/main/java/io/micronaut/security/oauth2/client/DefaultOpenIdProviderMetadata.java]: it's your repo, do whatever you think is better. the changes LGTM.

**Pull request merged**